component/azstorage/block_blob.go

```diff
@@ -649,6 +674,16 @@ func (bb *BlockBlob) calculateBlockSize(name string, fileSize int64) (blockSize
 	return blockSize, nil
 }
 
+func trackUpload(name string, bytesTransferred int64, count int64, uploadPtr *int64) {
+	if bytesTransferred >= (*uploadPtr)*100*1024*1024 || bytesTransferred == count {
+		(*uploadPtr)++
+		log.Debug("BlockBlob::trackUpload : Upload: Blob = %v, Bytes transferred = %v, Size = %v", name, bytesTransferred, count)
+
+		// send upload progress as event
+		AzStatsCollector.AddStats(compName, "UploadProgress", name, true, map[string]interface{}{"Bytes Transferred": bytesTransferred, "Size": count})
```

Comment:
We need to check how much perf hit we get due to these new stats and upload/download tracking